src/main/java/org/quantumbadger/redreader/settings/SettingsActivity.java

```diff
@@ -63,6 +65,8 @@ public void onBuildHeaders(final List<Header> target) {
 	public boolean onOptionsItemSelected(final MenuItem item) {
 		switch(item.getItemId()) {
 			case android.R.id.home:
+				if(PrefsUtility.appearance_navigation_type(this, sharedPreferences).equals("drawer_tabs"))
```

Comment:
This seems like it will leave all the user's previous activities running in the background, and open a new copy of MainActivity on top of them?